ci: add semantic-release

alexashley opened the pull request:
Also updates the README, as it was missing a few flags

alexashley commented on .github/workflows/build.yaml:
```diff
@@ -2,9 +2,7 @@ name: app
-      - '*'
-    tags:
-      - v*.*.*
+      - '**'
```
I noticed the workflow wasn't running on the dependabot branches. Dependabot creates branches with a `/` in the name, and the wildcard only matches up to the first path segment. If we were to turn off Dependabot and use Renovate, it also makes branches with a similar pattern. 

https://github.com/orgs/community/discussions/25445

rcoy-v commented:
This is using semantic release to just create a tag, which will start the separate build workflow with go releaser, correct? Could we, or does it make sense, to have semantic release also handle the release like the policy repository? https://github.com/liatrio/gh-trusted-builds-policy/blob/main/.releaserc#L21

alexashley commented:
> This is using semantic release to just create a tag, which will start the separate build workflow with go releaser, correct

It shouldn't trigger a new workflow run, I removed the tags trigger and changed the release job to always run on `main`. Goreleaser just runs after and it should have the local tag. 

The Goreleaser docs have an example of integrating with semantic-release but it's just using the exec plugin to run goreleaser. I don't see that as too different from just running the action, especially if we aren't going to create releases locally https://goreleaser.com/cookbooks/semantic-release/#example-semantic-release
